pkg/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2022 The Fission Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/fission/fission/pkg/controller/client"
+	"github.com/fission/fission/pkg/fission-cli/util"
+)
+
+type CategoryID string
+
+const (
+	Kubernetes      CategoryID = "kubernetes"
+	FissionServices CategoryID = "fission-services"
+	FissionVersion  CategoryID = "fission-version"
+)
+
+var (
+	okStatus   = color.New(color.FgGreen, color.Bold).SprintFunc()("\u221A") // √
+	failStatus = color.New(color.FgRed, color.Bold).SprintFunc()("\u00D7")   // ×
+)
+
+// Category is a group of checkers
+type Category struct {
+	ID       CategoryID
+	checkers []Checker
+	enabled  bool
+}
+
+type Checker struct {
+	successMsg string
+	check      func() error
+}
+
+type Options struct {
+	KubeContext   string
+	FissionClient client.Interface
+}
+
+type HealthChecker struct {
+	categories []*Category
+	*Options
+
+	kubeAPI          *kubernetes.Clientset
+	fissionNamespace string
+}
+
+func isCompatibleVersion(minimalRequirementVersion [3]int, actualVersion [3]int) bool {
+	if minimalRequirementVersion[0] < actualVersion[0] {
+		return true
+	}
+
+	if (minimalRequirementVersion[0] == actualVersion[0]) && minimalRequirementVersion[1] < actualVersion[1] {
+		return true
+	}
+
+	if (minimalRequirementVersion[0] == actualVersion[0]) && (minimalRequirementVersion[1] == actualVersion[1]) && (minimalRequirementVersion[2] <= actualVersion[2]) {
+		return true
+	}
+
+	return false
+}
+
+func (hc *HealthChecker) CheckKubeVersion() (err error) {
+
+	version, err := hc.kubeAPI.ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	major, _ := strconv.Atoi(version.Major)
+	minor, _ := strconv.Atoi(version.Minor)
+	apiVersion := [3]int{major, minor, 0}
+	minAPIVersion := [3]int{1, 19, 0}
```

Comment:
Minimum Kubernetes version required is hardcoded. I want to confirm whether this should go inside `pkg/apis/core/v1/const.go

Reply:
I agree we should have constants for Kubernetes version. 
We can consider https://github.com/fission/fission/blob/master/pkg/fission-cli/util/constants.go

Reply:
This seems better place `pkg/apis/core/v1/const.go` Lets use `pkg/apis/core/v1/const.go`